wasm_codec/derive/src/lib.rs

```diff
@@ -99,13 +143,32 @@ fn gen_output(
     }
 }
 
+/// [`TokenStream2`] wrapper which will be lazily evaluated
+///
+/// Implements [`quote::ToTokens`] trait
+struct LazyTokenStream<F>(once_cell::unsync::Lazy<TokenStream2, F>);
```

Comment:
Can you explain why do we need this?

Reply:
It's a workaround to keep the code in `gen_body()` more clear. I wrap `memory` and `alloc` retrieving into this structure. But this retrieving requires me to know the *state* type. And I don't need `memory` and `alloc` in all cases.

So in some cases I don't need any of them and this lazy code will never be evaluated. This gives us:
1. A little bit faster compliation (probably a very little decrease)
2. Checking for *state* type presence only when we need it (this is mainly why I did it)